lib/message.mli

```diff
@@ -0,0 +1,19 @@
+(** Messages received by the client, whether they're requests,
+responses, or protocol-specific messages. For consumer use
+only when implementing a routing function for the
+client. *)
+
+(** Messages are requests or responses,
+determining how they're stored and
+where they're handled *)
+type label =
```

Comment:
I am not totally convinced by this naming. I feel `message_type` would be better, what do you think?

Reply:
How about category? In the rest of the codebase the type will appear `Message.message_type` which I'd like to avoid 😅

Reply:
Ok for `category`.